### ISSUE TITLE
nimble-tree-view Add checkbox in tree view

### ISSUE THREAD

**Issue by vikisekarNI** [ni/nimble#472]
## 😯 Problem to Solve

User should be able to select or unselect items in a tree with a checkbox

## 💁 Proposed Solution

- Create a nimble tree with nimble checkbox support for each tree item
- Selecting parent and/or child selection needs to be handled. For ex. If we select parent, all its child should be selected and if we are selecting only few children, the parent node should go to an intermediate state

![TreeWithCheckbox](https://user-images.githubusercontent.com/94439533/161071860-b36818c7-3403-42b6-910e-3b4dc4fb19a9.gif)

**Comment by nate-ni** [comment]
Is this till blocked by FAST or does it now support multi-select. Is there something we can do to enable this?

**Comment by jattasNI** [comment]
The client who originally requested this [was able to implement the behavior they needed themselves](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/Web/Workspaces/TestInsights/src/app/dataspaces/components/axis-selection-slideout/axis-selection-slideout.component.html&version=GBmaster&line=11&lineEnd=12&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) by composing a `nimble-checkbox` inside a `nimble-tree-item`. I propose we keep this issue around since it would be better to for Nimble to provide the multiselect behavior someday. But it's not a priority now to implement it ourselves or ask FAST for it. That priority could change if designers or clients request it again.

**Comment by jattasNI** [comment]
Removing the 'blocked' label because we could implement this in Nimble by forking the FAST template as we've done for other components. I still don't think it's a high priority for reasons mentioned above.

**Comment by fredvisser** [comment]
Closing as low ROI
